drafts/pair-programming-mistakes.md

```diff
@@ -0,0 +1,66 @@
+
+
+## The Trap of No Breaks
+Despite being aware of the importance of breaks and switch-ups, I often find myself falling into the trap of continuous pairing. When [I’m in the flow](https://en.wikipedia.org/wiki/Flow%5F(psychology)#:~:text=The%20flow%20state,abilities.%22%5B4%5D), time and the need for rest seems to slip away unnoticed, which also happens often when pairing. The driver does not switch with the navigator after a committable segment, and we continue to push forward. Contrary to popular belief, [being in the flow isn’t always advantageous](https://en.wikipedia.org/wiki/Flow%5F(psychology)#:~:text=In%20some%20cases,on%20attentional%20abilities.), since it can lead to fatigue and overworking. I noticed that when we are tired, the quality and quantity of our communication rapidly decreases. We all start to think about our own problem and ignore our partner, or we are stuck on the wrong problem, a phenomenon known as the [XY problem](https://en.wikipedia.org/wiki/XY%5Fproblem). This significantly reduces our productivity and may even render us ineffective.
+
+![Tired of working non-stop.](https://cdn.hashnode.com/res/hashnode/image/upload/v1717162596435/FRNbO6MkS.png?auto=format)
+
+The key in avoiding this trap is to remember to take regular breaks, switch roles and ensure effective communication at all times. This not only helps to keep our minds fresh but also ensures that we are on the same page throughout the whole session.
+
+
+## Remote Pairing under Non-Ideal Conditions
+There was an instance when a colleague and I attempted to pair program remotely. We experienced initially minor, but with time major delay, which caused us frustration and irritation. However, not wanting to constantly complain, we tried to cope with the situation.
+
+![Issues when working remotely.](https://cdn.hashnode.com/res/hashnode/image/upload/v1717162575632/jl%5F2eYy7A.png?auto=format)
+
+Unfortunately, the delay made it nearly impossible to follow the driver’s actions, and we eventually had to stop the pairing session. This experience underscored the importance of having an excellent setup and high-quality connection for remote pair programming. Even minor issues can become major distractions, diverting focus from the task at hand.
```

Comment:
This experience underscored the importance of having an excellent setup and high-quality connection for remote pair programming

This experience underscored the importance of having an excellent setup and a high-quality internet connection for remote pair programming